weasyl/searchtag.py

```diff
@@ -217,3 +268,170 @@ def tag_history(submitid):
+def get_user_searchtag_blacklist(userid):
+    """
+    Retrieves a list of tags on the user searchtag blacklist for friendly display to the user.
+
+    Parameters:
+        userid: The userid of the user requesting the list of tags.
+
+    Returns:
+        A list of blacklisted searchtag titles which were set by ``userid``.
+    """
+    query = d.engine.execute("""
+        SELECT st.title
+        FROM searchmapuserblacklist
+        INNER JOIN searchtag AS st USING (tagid)
+        WHERE userid = %(userid)s
+        ORDER BY st.title ASC
+    """, userid=userid).fetchall()
+    tags = [tag.title for tag in query]
+    return tags
+
+
+def get_global_searchtag_blacklist(userid):
+    """
+    Retrieves a list of tags on the global searchtag blacklist for friendly display to the director.
+
+    Parameters:
+        userid: The userid of the user requesting the list of tags.
+
+    Returns:
+        A list of globally blacklisted searchtag titles and the name of the user which added it.
+    """
+    # Only directors can view the global blacklist; sanity check against the @director_only decorator
+    if userid not in staff.DIRECTORS:
+        raise WeasylError("InsufficientPermissions")
+
+    query = d.engine.execute("""
+        SELECT st.title, lo.login_name
+        FROM searchmapglobalblacklist
+        INNER JOIN searchtag AS st USING (tagid)
+        INNER JOIN login AS lo USING (userid)
+        ORDER BY st.title ASC
+    """).fetchall()
+    return query
+
+
+def query_blacklisted_tags(ownerid, newtagids):
+    """
+    Checks both the user and global searchtag blacklists against added tags, and
+        returns the IDs of any tags matching the STBL, either based on strict
+        match, or regexp based match.
+
+    Parameters:
+        ownerid: The userid of the submitted item being checked.
+        newtagids: The list of added tag ids.
+
+    Returns:
+        blacklisted_tags: The tagids which are blacklisted as a set()
+    """
+    blacklist_query = d.engine.execute("""
+        SELECT st.tagid, st.title
+        FROM searchmapuserblacklist
+        INNER JOIN searchtag AS st USING (tagid)
+        WHERE userid = %(ownerid)s
+        UNION
+        SELECT st.tagid, st.title
+        FROM searchmapglobalblacklist
+        INNER JOIN searchtag AS st USING (tagid)
+    """, ownerid=ownerid).fetchall()
+    tag_titles = d.engine.execute("""
+        SELECT title, tagid
+        FROM searchtag
+        WHERE tagid = ANY (%(tagids)s)
+    """, tagids=list(newtagids)).fetchall()
+
+    blacklisted_tag_ids = set()
+
+    for x in blacklist_query:
+        # Determine if the candidate ID is directly present in the newly added IDs.
+        if x.tagid in newtagids:
+            blacklisted_tag_ids.add(x.tagid)
+        # Otherwise we need to parse for wildcards
+        elif "*" in x.title:
+            # Convert '*' to '.*' as expected by regexp, and add start|end anchors
+            regex =  "^" + x.title.replace("*", ".*") + "\Z"
```

Comment:
Only one space after `=`. Also, you can use string formatting `r"^%s\Z" % (…,)`, or drop the starting anchor (`match` only matches at the start of a string).